lib/r10k/tarball.rb

```diff
@@ -0,0 +1,257 @@
+require 'digest'
+require 'fileutils'
+require 'find'
+require 'minitar'
+require 'net/http'
+require 'tempfile'
+require 'uri'
+require 'zlib'
+require 'r10k/settings'
+require 'r10k/settings/mixin'
+require 'r10k/util/platform'
+
+module R10K
+  class Tarball
+
+    include R10K::Settings::Mixin
+
+    def_setting_attr :proxy
+    def_setting_attr :cache_root, if R10K::Util::Platform.windows?
+                                    File.join(ENV['LOCALAPPDATA'], 'r10k', 'tarball')
+                                  else
+                                    File.expand_path(ENV['HOME'] ? '~/.r10k/tarball': '/root/.r10k/tarball')
```

Comment:
Would it make sense to just do
```suggestion
                                    File.join(ENV['HOME'] || '/root', '.r10k', 'tarball')
```
?

Reply:
Do we need to account for `ENV['LOCALAPPDATA']` being unset?

Reply:
Also, is there a reason to have a different `cache_root` from other module types? I'm looking at https://github.com/puppetlabs/r10k/blob/main/lib/r10k/forge/module_release.rb#L16 and https://github.com/puppetlabs/r10k/blob/main/lib/r10k/forge/module_release.rb#L66 and wondering how much of that can be reused.

Reply:
The user setting `cachedir` will dictate what is used here, just as it does [for git and forge cache roots](https://github.com/puppetlabs/r10k/blob/a722b71f49c8e8249f978ac443eb78afe877796f/lib/r10k/initializers.rb#L36-L38).

The value supplied is the default, and is [modeled after R10K::Git::Cache](https://github.com/puppetlabs/r10k/blob/a722b71f49c8e8249f978ac443eb78afe877796f/lib/r10k/git/cache.rb#L19-L29). That model does not account for `ENV['LOCALAPPDATA']` being unset.

It would be cleaner overall if there were a central way to define this default, but at present it doesn't seem like there is one. I think figuring out how to change that may best be done as a separate refactor PR.

Reply:
Note: stumbled across settings.rb, and it looks like there may be a way to set defaults there. Probably worth reviewing. If so, then the default can be set there, and these classes (`R10K::Tarball`, `R10K::Git::Cache`) shouldn't need to set a default themselves... 🤔